### PR TITLE
[MI-2783] Integrate button to disconnect connected channels

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

**Review by raghavaggarwal2308 — APPROVED** [review]

File: src/types/index.d.ts [hunk 1]
```diff
@@ -4,16 +4,16 @@ type APIError = {
...
-    TeamsChannelName: string,
-    MMTeamName: string,
-    MMChannelID: string,
-    MMChannelName: string,
-    MMUserID: string,
+    subscriptionId: string,
+    clientState: string,
+    teamsTeamId: string,
+    teamsTeamName: string,
+    teamsChannelID: string,
```
Comment (Kshitij-Katiyar) [inline src/types/index.d.ts]: maybe ?
```suggestion
    teamsChannelId: string,
```

File: src/types/index.d.ts [hunk 2]
```diff
@@ -4,16 +4,16 @@ type APIError = {
...
+    teamsChannelName: string,
+    mmTeamName: string,
+    mmChannelId: string,
+    mmChannelName: string,
+    mmUserID: string,
```
Comment (Kshitij-Katiyar) [inline src/types/index.d.ts]: same

File: src/containers/HomePage/tablePanel.tsx [hunk 3]
```diff
@@ -1,15 +1,19 @@
...
+import {useDisconnectChannelMutation, useGetConnectedChannelsQuery} from '../../services';
 import {APIError, ConnectedChannelTableData} from '../../types';
 
-import {useGetConnectedChannelsQuery} from '../../services';
-
 const TablePanel = () => {
     // States
     const [rowsData, setRowsData] = useState<ConnectedChannelTableData[]>([]);
+    const [showResultPanel, setShowResultPanel] = useState(false);
+    const [subsriptionDeletedId, setSubsriptionDeletedId] = useState('');
```
Comment (avas27JTG) [inline src/containers/HomePage/tablePanel.tsx]:
```suggestion
    const [deletedSubsriptionId, setDeletedSubsriptionId] = useState('');
```

File: src/containers/HomePage/tablePanel.tsx [hunk 4]
```diff
@@ -57,18 +65,27 @@ const TablePanel = () => {
...
-                                    secondary={true}
-                                    content='Disconnect'
-                                    onClick={() => handleDisconnectButton(row.TeamsChannelName, row.MMChannelName)}
-                                />
+                                <>
+                                    <Button
+                                        secondary={true}
+                                        content='Disconnect'
+                                        onClick={() => handleDisconnectButton(row.subscriptionId)}
+                                        disabled={disconnectChannelIsLoading && subsriptionDeletedId === row.subscriptionId}
```
Comment (avas27JTG) [inline src/containers/HomePage/tablePanel.tsx]: ### 
```suggestion
                                        disabled={isDisconnectChannelLoading && subsriptionDeletedId === row.subscriptionId}
```

File: src/containers/HomePage/tablePanel.tsx [hunk 5]
```diff
@@ -78,7 +95,14 @@ const TablePanel = () => {
 
             setRowsData(rows);
         }
-    }, [data, isError, isFetching]);
+    }, [data, isError, isFetching, disconnectChannelIsLoading]);
+
+    useEffect(() => {
+        if (disconnectChannelIsSuccess || disconnectChannelIsError) {
```
Comment (avas27JTG) [inline src/containers/HomePage/tablePanel.tsx]:
```suggestion
        if (disconnectChannelIsSuccess || isDisconnectChannelError) {
```

Comment (avas27JTG) [inline src/containers/HomePage/tablePanel.tsx]:
```suggestion
        if (isDisconnectChannelSuccess || disconnectChannelIsError) {
```

**Review by avas27JTG — APPROVED** [review]

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**